airflow/hooks/hdfs_hook.py

```diff
@@ -32,19 +32,30 @@ def get_conn(self):
         '''
         Returns a snakebite HDFSClient object.
         '''
+        connections = self.get_connections(self.hdfs_conn_id)
+
         use_sasl = False
         if configuration.get('core', 'security') == 'kerberos':
             use_sasl = True
 
-        connections = self.get_connections(self.hdfs_conn_id)
         client = None
-	''' When using HAClient, proxy_user must be the same, so is ok to always take the first '''
-	effective_user = self.proxy_user or connections[0].login
+
+        ''' When using HAClient, proxy_user must be the same, so is ok to always take the first '''
+        effective_user = self.proxy_user or connections[0].login
         if len(connections) == 1:
-            client = Client(connections[0].host, connections[0].port, use_sasl=use_sasl, effective_user=effective_user)
+            autoconfig = connections[0].extra_dejson.get('autoconfig', 'false')
+            if autoconfig == 'true':
```

Comment:
json supports booleans, so you may want to simply `if autoconfig:`, or more explicitely `if autoconfig is True or autoconfig.lower() == 'true` maybe